refactor(transformer/typescript): reuse `Atom`

overlookmotel opened the pull request:
Re-use existing `Atom` rather than allocating multiple copies of it into the arena.

Dunqing commented:
### Merge activity

* **Dec 17, 9:30 PM EST**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

overlookmotel committed:
refactor(transformer/typescript): reuse `Atom` (#7969)

Re-use existing `Atom` rather than allocating multiple copies of it into the arena.
